### ISSUE TITLE
All the information i have collected regarding BYOND's overlay-related graphics bugs

### ISSUE THREAD

**Issue by Tacolizard** [tgstation/tgstation#22894]
the whole parallax bug thing seems related to an issue on certain computers (mabye only laptops) where blend modes are broken. and old bug has been around for a while on goonstation, where their  new lighting system rendered wrong for some people, causing a bunch of rainbow-colored boxes to overlay everything. through testing i found that anyone who had this happen to them also experienced glitches with our parallax space, whereby only the bottom layer was visible. i think this has to do with BLEND_ADD, given that the second parallax layer uses that and so does parts of goonlights. **this issue is not explicity related to software rendering, given that someone's laptop running native windows with a dedicated gpu experienced the glitch.** i also read MSO's byond bug report about having a dual-gpu laptop caused similar issues, however this problem happens even on single-gpu, cpu rendering, laptops. 

Interestingly, I have been experimenting with a number of OSes on my mac laptop. I have integrated graphics only. i have found that:

1. running a VM on OSX causes graphics issues
2. Running a VM on a linux distro without Mesa3D causes the issues
3. if on linux, installing Mesa3D solves the issue
4. The issue is independent of which VM software is used

in conclusion: 

it seems that this is a pretty massive graphics bug, that involves bugs in BYOND in conjunction with certain hardware setups and graphics drivers.

**Comment by optimumtact** [comment]
See also #21264

possibly also #2601

**Comment by optimumtact** [comment]
What I really would like to hear from is people with *native* windows installations who encounter this.

So far everyone I have spoken to was doing some form of VM related activity without native hardware passthrough

**Comment by Firecage** [comment]
oranges, does it count if I say that sometimes when I log on, almost all sprites(including characters, except their hair overlay) are square 32x32 black boxes and I have to relog to fix it?

**Comment by MrStonedOne** [comment]
This is what it always looks like when graphics hardware rendering fails.

Plain and simple, you need to have working graphics acceleration to use planemaster features, and we do quite a few, and more will be coming as planemasters allow us to do selective visibility without client images and that means a lot of shit code can soon be removed, and quite a few systems can be improved.

If there are any tricks to get this to work on vms, that should be the route we should go down.

As well as helping byond fix bugs that break hardware rendering.

**Comment by MrStonedOne** [comment]
It's either that or we declare planes and planemasters off limits and loose all the tricks we could do with that

**Comment by Tacolizard** [comment]
@optimumtact I had windows native installed on my mac for a little while, using bootcamp drivers to support my mac hardware. no graphics issues. The laptop i mentioned with a dual GPU was sold with windows.

**Comment by Tacolizard** [comment]
@optimumtact #21264 seems to be a similar issue, but everyone i talked to regarding lighting blending and parallax had no problems other than lighting and parallax.

**Comment by Tacolizard** [comment]
one last thing, if it helps; i've used some of the lighting system library demos on the byond developer hub, and lighting systems that used smoothing such as FIREking's "Light Speed" displayed/rendered perfectly for me.

**Comment by Tacolizard** [comment]
@optimumtact >this issue is not related to software rendering, given that someone's laptop running native windows with a dedicated gpu experienced the glitch.

**Comment by Tacolizard** [comment]
@MrStonedOne The thing is, this is probably a BYOND bug. I think this because without graphics acceleration even more graphical problems happen, as expected. Wouldn't this mean that graphics acceleration must work at some level? The issue must be related to how BYOND uses graphics acceleration for certain features, because some features that require acceleration work fine.

**Comment by Tacolizard** [comment]
@optimumtact AFAIK, there is no way to do hardware passthrough of an Intel CPU's integrated graphics on a VM

**Comment by optimumtact** [comment]
well no you can't, it always has to be discrete graphics

**Comment by Tacolizard** [comment]
@optimumtact @MrStonedOne FUCK YEAH BIATCH I FIGURED IT OUT I SOLVED THE HIGGS BUGSON.


to fix all parallax glitches+make goonlighting/vglighting work, **you just need to install the June 2010 DirectX SDK**. That's fucking it.

tested on the latest BYOND Beta Build 511.1367,  though it probably works on the stable version too.

**Comment by PKPenguin321** [comment]
nice

**Comment by optimumtact** [comment]
Hmm that's a very interesting update, thank you

**Comment by clusterfack** [comment]
You guys didn't know about this? You could have just asked us we would have told you.

**Comment by optimumtact** [comment]
Gee thanks cluster

**Comment by SheepFighter** [comment]
Is there a way to do this for Windows 10, or am I borked?

**Comment by SheepFighter** [comment]
Nevermind, I got it!

**Comment by optimumtact** [comment]
@SheepFighter do report in if it fixes it for you, along with any links you found along the way

**Comment by Tacolizard** [comment]
@SheepFighter @optimumtact im pretty sure you can install the june 2010 sdk on any windows from XP forwards
